### PR TITLE
Bugfix/iia 897 exception on new pattern and add details for fields section

### PR DESCRIPTION
**Pull request by swaroopsalvi**

### CODE REVIEW

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternViewModel.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public PatternViewModel() {
                     .addProperty(FQN_DESCRIPTION_NAME_TEXT, "")
                     .addProperty(OTHER_NAME_DESCRIPTION_NAME_TEXT, "")
                     // Ordered collection of Fields
-                    .addProperty(FIELDS_COLLECTION, (List<PatternField>) null)
+                    .addProperty(FIELDS_COLLECTION, (List<PatternField>) new ArrayList<PatternField>())
```
[the remark this answers is not in the document]
Comment (swaroopsalvi) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternViewModel.java:86]: The null return will throw NULL Pointer exception in the PatternDetailsController.java: in the following logic. Hence I am initializing it to ArrayList.

        patternFieldsPanelEventSubscriber = evt -> {
            List<PatternField> patternFieldList = patternViewModel.getObservableList(FIELDS_COLLECTION);
            patternFieldList.add(evt.getPatternField());
            addField(evt.getPatternField());
        };

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java [hunk 2]
```diff
@@ -413,12 +417,20 @@ private void generateMeaningForm() {
...
+        removeMeaning();
         patternFieldsViewModel.save(true);
     }
 
     @FXML
     private void onCancel(ActionEvent actionEvent) {
+        //publish close env
+        EvtBusFactory.getDefaultEvtBus().publish(patternFieldsViewModel.getPropertyValue(PATTERN_TOPIC),
+                new PatternPropertyPanelEvent(actionEvent.getSource(), CLOSE_PANEL));
+        clearView(actionEvent);
```
Comment (dholubek) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java:433]: is cancel supposed to call clearView? which in turn removes the purpose and meaning?  I thought Sundeep mentioned that cancel isn't supposed to clear... is that right?

**Review by dholubek — COMMENTED** [review]
had a question on cancel and clear

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java (hunk 2 above)
Comment (swaroopsalvi) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java:433]: As per my understanding, Cancel and Done both should clearView. Based on what I was told, every time a user clicks on the pencil icon a clear form should be presented.

Comment (dholubek) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternFieldsController.java:433]: you are correct, will approve and merge

**Review by dholubek — APPROVED** [review]

**Pull request merged**